src/rules/order.js

```diff
@@ -858,9 +858,12 @@ module.exports = {
         properties: {
           groups: {
             type: 'array',
+            // Verified manually in the convertGroupsToRanks function
```

Comment:
no need for the comment. but this can be verified to have numbers in it, and no duplicates, yes?

Reply:
The [secondary validation](https://github.com/import-js/eslint-plugin-import/blob/a8ab6461544c3d908df3dafaccff823a29e683fe/src/rules/order.js#L599-L604) has its own error messages. I believe you're saying to move the duplicates/existence validation logic up to the schema? That would change the error messages and perhaps other behavior. For instance, there are currently [unit tests](https://github.com/import-js/eslint-plugin-import/blob/4f145a2c64af4931f4bf3ae951c8b719b544718f/tests/src/rules/order.js#L1646-L1702) that check for the current error messages.

Would that count as tightening the schema of a released option?

Reply:
Changed error messages are fine; test cases that always error, changing to "removed test cases because it's now unreachable" is also fine, I think.

Reply:
Gotcha, I'll take a look